scripts/upsert_mgrb_participants.py

```diff
@@ -0,0 +1,177 @@
+import csv
+import logging
+from collections import defaultdict
+from io import StringIO
+from typing import Dict, List
+
+import click
+from google.cloud import storage
+
+from metamist.apis import ParticipantApi
+from metamist.graphql import gql, query
+from metamist.models import (
+    ParticipantUpsert,
+    SampleUpsert,
+)
+
+BUCKET_NAME = 'cpg-mgrb-test-upload'
+
+SAMPLE_QUERY = gql(
+    """
+    query getSampleData($project: String!) {
+        project(name: $project) {
+            id
+        }
+        sample(project: {eq: $project}) {
+            id
+            active
+            externalId
+            sequencingGroups {
+            id
+            assays {
+                id
+                meta
+            }
+            analyses {
+                id
+                output
+                outputs
+                meta
+                }
+            }
+        }
+    }
+    """
+)
+
+PARTICIPANT_QUERY = gql(
+    """
+    query getParticipantData($project: String!) {
+    project(name: $project) {
+        participants {
+            id
+            externalId
+            }
+        }
+    }
+    """
+)
+
+papi = ParticipantApi()
+
+
+def map_sample_eid_to_sample_data(data):
+    """Map sample external ID to sample data"""
+    return {sample['externalId']: sample for sample in data['sample']}
+
+
+def get_sample_data(project):
+    """Get sample data from metamist"""
+    response = query(SAMPLE_QUERY, {'project': project})
+    if not isinstance(response, dict):
+        raise ValueError('Expected a dictionary response from query')
+    return response
+
+
+def create_participant(sample_data: list, peid: str, sex: int):
+    """Create participant upsert object"""
+    participant_upsert = [
+        ParticipantUpsert(
+            id=None,
+            external_ids={'': peid},
```

Comment:
Why do we pass an empty string here? I think it would be beneficial to call it something.

Reply:
My understanding was, it has to be empty to becomes the main organisation external_id:
https://github.com/populationgenomics/metamist/blob/ab02bd2190107ca39db73b61b415a86960452ff7/models/base.py#L13